contracts/colony/ColonyExpenditure.sol

```diff
@@ -174,6 +175,43 @@ contract ColonyExpenditure is ColonyStorage {
     expenditureSlots[_id][_slot].claimDelay = _claimDelay;
   }
 
+  uint256 constant EXPENDITURES_SLOT = 25;
+  uint256 constant EXPENDITURESLOTS_SLOT = 26;
+  uint256 constant EXPENDITURESLOTPAYOUTS_SLOT = 27;
+
+  function setExpenditureState(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _id,
```

Comment:
It needs to be its own argument, because otherwise how else could we evaluate the domain proof? But I do think we should check that the expenditure exists.

Reply:
`expenditures[_keys[0]].domainId` would do the job, no?

Reply:
Hmm... I guess there's no way for a user to get somewhere they shouldn't by passing a wrong id. Any bad arguments would just lead to writing a nonsense slot.

Reply:
Thinking about this more -- I don't think it makes as much sense to force `expenditureId` as the first argument, since all it does is then give us a second job of enforcing that the first argument of `mask` is a mapping. If we know in advance what a value needs to be, it should be its own argument. The keys should be allowed to take any value they need to.